docs/design/libraries/DllImportGenerator/Compatibility.md

```diff
@@ -71,6 +71,8 @@ Only single-dimensional arrays are supported for source generated marshalling.
 
 In the source-generated marshalling, arrays will be allocated on the stack (instead of through [`AllocCoTaskMem`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.marshal.alloccotaskmem)) if they are passed by value or by read-only reference if they contain at most 256 bytes of data. The built-in system does not support this optimization for arrays.
 
+In the built-in system, marshalling a `char` array by value with `CharSet.Unicode` would default to also marshalling data out. In the source-generated marshalling, the `char` array must be marked with the `[Out]` attribute for data to be marshalled out by value.
```

Comment:
This is due to the fact that a char array with CharSet.Unicode used to be pinnacle as char is blittable with CharSet.Unicode

Reply:
Yeah, I figured that was it once I saw the behaviour, but it was not something I realized immediately 😕